Sources/swiftpm-xctest-helper/main.swift

```diff
@@ -0,0 +1,120 @@
+    dup2(newFd, 1)
+    close(newFd)
+
+    let result = try body()
+
+    fflush(stdout)
+    dup2(oldFd, 1)
+    close(oldFd)
+
+    return result
+}
+
+extension String {
+    func normalizedPath() -> String {
+        var path = self
+        if !(path as NSString).isAbsolutePath {
+            path = NSFileManager.default().currentDirectoryPath + "/" + path
+        }
+        return (path as NSString).standardizingPath
+    }
+}
+
+extension NSFileHandle {
+    func writeUTF8String(_ string: String) throws {
+        guard let data = (string as NSString).data(using: NSUTF8StringEncoding) else {
+            throw Error.couldNotConvertStringToNSData(string)
+        }
+        write(data)
+    }
+}
+
+do {
+    guard Process.arguments.count == 3 else {
+        throw Error.invalidUsage
+    }
+    let bundlePath = Process.arguments[1].normalizedPath()
+    let outputFile = Process.arguments[2].normalizedPath()
+
+    // Bundle may write to stdout when it is being loaded, redirect that output to /dev/null.
+    let suite: XCTestSuite = try withRedirectedStdout {
+        guard let bundle = NSBundle(path: bundlePath) where bundle.load() else {
+            throw Error.unableToLoadBundle(bundlePath)
+        }
+        return XCTestSuite.default()
+    }
+    
+    let splitSet: Set<Character> = ["[", " ", "]", ":"]
+
+    // Create output file.
+    NSFileManager.default().createFile(atPath: outputFile, contents: nil, attributes: nil)
+
+    // Open output file for writing.
+    guard let file = NSFileHandle(forWritingAtPath: outputFile) else {
+        throw Error.couldNotOpenOutputFile(outputFile)
+    }
+
+    for case let testCaseSuite as XCTestSuite in suite.tests {
+        for case let testCaseSuite as XCTestSuite in testCaseSuite.tests {
+            for case let test as XCTestCase in testCaseSuite.tests {
+                // Extract module name using reflection. Reflection format: `ModuleName.ClassName`.
+                let moduleName = String(String(reflecting: test.dynamicType).characters.split(separator: ".").first!)
+
+                // Split the test description into an array. Description formats:
+                // `-[ClassName MethodName]`, `-[ClassName MethodNameAndReturnError:]`
+                let exploded = test.description.characters.split(isSeparator: splitSet.contains).map(String.init)
+                let className = exploded[1]
+                var methodName = exploded[2]
+
+                // Unmangle names for Swift test cases which throw.
+                if methodName.hasSuffix("AndReturnError") {
+                    methodName = methodName[methodName.startIndex..<methodName.index(methodName.endIndex, offsetBy: -14)]
+                }
+                let outputString = "\(moduleName).\(className)/\(methodName)"
+                try file.writeUTF8String(outputString)
+                try file.writeUTF8String("\n")
+            }
+            try file.writeUTF8String("\n")
```

Comment:
I think we should write the data as JSON, just so it is a trivial schema to load. See some comments:
  https://github.com/apple/swift-corelibs-xctest/pull/114
for some thoughts on the schema.